### DISCUSSION TITLE
DotNetIsolator in Oqtane 5 ?

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#3411] [General] [answered]
Hi,

Is it useful / worth to include DotNetIsolator in Oqtane 5 ?

**More info:** https://github.com/SteveSandersonMS/DotNetIsolator

**Comment by thabaum** [comment]
@Sharafudeen This is very interesting and looks like something a developer could create a module to do from within Oqtane Framework if needed.  It has no network or data connectivity as it is isolated into a WebAssembly sandbox. I am not sure there is a direct use case for it within the Oqtane Framework and it looks like it is also still "experimental" and with no security testing done.

Is this part of a .NET 8 as I have heard nothing about it?

Can you think of any use cases for this from within the Oqtane Framework?

Maybe .NET 9 will introduce this if it isn't in 8 to explore further.

**Reply by Rodien** [reply]
Won't this fix the DLL hell issue?
Where DLL versions clash with each other.

**Reply by thabaum** [reply]
It looks like it is isolated, so nothing touches it,  you cant talk to Oqtane Database either.  Everything will be contained within a container more or less that has no network connectivity whatsoever.  What is built with the DotNetIsolator package is confined to it's code within.  

No data connectivity or network connectivity, and my guess is it more or less sounds like static web pages 
app"  with a built in server to produce the WebAssembly pages.

Oqtane is dynamic and relies heavily on a database of some kind in other words if you can't use a database how do you route pages and modules with the Oqtane Framework?  Oqtane wont be able to communicate with this, nothing can, and it cant communicate with anything either (no network connectivity).  

Maybe I am misunderstanding it when it says "no data connectivity" as in no external data connectivity to files outside of what is inside the instance, but it can connect to any data files that are within it's instance?  However maybe there are no "ports" or "ip address" or "domain name resolution" which limits its abilities to communicate with anything OR maybe you can supply it with a database like MongoDb inside the instance of it running or some file based data storage to interact with?

As I said I believe you can make a module within Oqtane that uses it and opens this isolated instance of Dot Net WebAssembly with some static information.   Might be good for displaying basic or private information once tested to be considered secure possibly?  But as it states no security tests have been performed to test if it is in fact a secure way to load a static page based app.

I imagine Oqtane could fire one of these instances up for you, but Oqtane itself would not be able to run on it unless you used some file based data like mongoDB or file based data storage?  Who really knows without pushing it's potential through a lot of trial and error testing.

This is my thoughts just reviewing what was presented by this again "Experimental" and "Not to be trusted" repo.

It does say "experimental" so have some fun with it share your results I would be interested in hearing what it's limitations are and potential use cases.

**Reply by Sharafudeen** [reply]
@thabaum We understood that DotNet Isolator is in “Experimental” stage and it has no guarantee in security. But when I’m looking demo session https://www.youtube.com/live/gxplZcW5t3Y?si=CKSgSgCQJibK9emW .. I’m so impressed and it has greater plans for WASI..

**Comment by sbwalker** [comment, marked as answer]
@Sharafudeen based on Steve Sanderson's explicit warning in the DotNetIsolator repo, we would definitely not include this in Oqtane 5:

"This is experimental and unsupported. It may or may not be developed any further. There will definitely be functional gaps. There are no guarantees about security."

This project was demonstrated by Steve Sanderson during the last MVP Summit to showcase some of his WebAssembly R&D efforts.
